### PR TITLE
Fix mario_rl_tutorial.py

### LINKED ISSUE #2335: Update tutorial to avoid use of copy.deepcopy()-Train a Mario-playing RL Agent
Update the  [Mario Playing RL Agent code](https://github.com/pytorch/tutorials/blob/6ae56a1a2639308550a69c5006e879d331f05b6d/intermediate_source/mario_rl_tutorial.py#L437) to avoid the use of copy.deepcopy().

Tutorial link: https://github.com/pytorch/tutorials/blob/6ae56a1a2639308550a69c5006e879d331f05b6d/intermediate_source/mario_rl_tutorial.py#L437

See issue #2177  for more information

cc @albanD @svekars @carljparker @NicolasHug @kit1980 @subramen

### PR DESCRIPTION
**Pull request by alperenunlu**
Fixes #2335

## Description

Removed copy.deepcopy method with appropriate torch methods on intermediate_source/mario_rl_tutorial.py file

## Checklist

- [x] The issue that is being fixed is referred in the description
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @vmoens @nairbv @albanD @sekyondaMeta @svekars @carljparker @NicolasHug @kit1980 @subramen

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
That makes more sense to me too, thanks for this.
